lib/galaxy/managers/tools.py

```diff
@@ -110,6 +110,8 @@ def create_tool(self, trans, tool_payload, allow_load=True):
                 tool_path=tool_path,
                 tool_directory=tool_directory,
                 uuid=uuid,
+                active=tool_payload.get("active", True),
```

Comment:
Would there be a use case when we want to specify `False` for either one of these? Their default values are `True`. And if we do, we probably don't want to specify the default value here and include it only if it has a value, otherwise rely on the database to supply the default.

Reply:
Sorry, I do not get the point of your question? Do you mean it's redundant to default the get() to True? If we only do get("active") then it will return None I suppose. Is the following DAO layer able to translate None to the default value then?

Reply:
BTW, out of curiosity ... I've been digging into this dynamic tool part because I need to dynamically generate tools to display in the tool panel according to the logged user (and only for the session), transforming data fetched from an external service and without rebooting the galaxy server.
Is this a use case meant for these dynamic tools? Are there any ways of doing this with the current implementation of Galaxy? Thank you.

Reply:
Just to further explain ... I was hoping that by setting hidden to false the tools would have appeared in the tool panel but I've noticed that it probably does not work like that...

Reply:
Yeah, so far they're only meant (and tested) for running jobs and workflows. You could probably implement a selector that looks at `/api/dyanmic_tools` to list these, but note that this functionality is also restricted to admins.

Reply:
Any ref to the code for selectors? Maybe I could write one such selector to populate the tool panel with ephemeral tools generated on the fly for the logged user?

Reply:
Thinking about that, you could probably stuff this in as a custom panel view in https://github.com/galaxyproject/galaxy/blob/33011b28cfcf20c54baf69a4040d3404cda11ebb/lib/galaxy/tool_util/toolbox/base.py#L1334-L1351 where you could define a custom view like `__dyanmic_tools__`. Then you'd have to serialize the dynamic tools in a structure like
```json
{
  "__dynamic_tools__": {
    "model_class": "ToolSection",
    "id": "__dynamic_tools__",
    "name": "Dynamic Tools",
    "version": "",
    "description": null,
    "links": null,
    "tools": [
      "$dynamic_tool_id_1"
    ]
  }
}
```

That'll make these tools selectable here:

<img width="408" alt="Screenshot 2024-05-03 at 13 31 56" src="https://github.com/galaxyproject/galaxy/assets/6804901/edec2034-f7cc-494b-a945-3589cacb939e">


Before embarking on that I'd also check that the tool form is rendered correctly for dynamic tools (at https://usegalaxy.org/?tool_id=$tool_id).

Reply:
> Sorry, I do not get the point of your question? Do you mean it's redundant to default the get() to True? If we only do get("active") then it will return None I suppose. Is the following DAO layer able to translate None to the default value then?

Sorry, I made it confusing! So yes, it is redundant to specify `True` as the default value here since it is already specified in the database. The potential problem is that if at some point we change the default, we'll have to remember to change it in both places, otherwise that would introduce a bug. My suggestion was to include these arguments in the call to `self.create` only if they have values, otherwise let the database handle the default. However, it's even simpler than that - include them regardless, just don't set the default value in `get`: it returns `None` by default, in which case the database will use `true`, by default, when inserting the row.

Reply:
Done .... pipelines restarting ...